benchmarks/TF/linux-x86_64.cmake

```diff
@@ -16,122 +16,81 @@
-    "iree-llvm-cpu-sync"
-  DRIVER
-    "local-sync"
-)
+  # CPU, LLVM, local-sync, x86_64, full-inference
+  iree_benchmark_suite(
+    GROUP_NAME
+      "linux-x86_64"
 
-# CPU, LLVM, local-task, 1 thread, x86_64, full-inference
-iree_benchmark_suite(
-  GROUP_NAME
-    "linux-x86_64"
+    MODULES
+      "${MINILM_L12_H384_UNCASED_INT32_SEQLEN128_MODULE}"
+      "${RESNET50_TF_FP32_MODULE}"
+      "${BERT_FOR_MASKED_LM_FP32_SEQLEN512_MODULE}"
+      "${EFFICIENTNET_V2_S_TF_FP32_MODULE}"
 
-  MODULES
-    "${MINILM_L12_H384_UNCASED_INT32_SEQLEN128_MODULE}"
-    "${RESNET50_TF_FP32_MODULE}"
-    "${BERT_FOR_MASKED_LM_FP32_SEQLEN512_MODULE}"
-    "${EFFICIENTNET_V2_S_TF_FP32_MODULE}"
+    BENCHMARK_MODES
+      "${BENCHMARK_MODES}"
+    TARGET_BACKEND
+      "llvm-cpu"
+    TARGET_ARCHITECTURE
+      "${TARGET_ARCHITECTURE}"
+    COMPILATION_FLAGS
+      ${LINUX_X86_64_CPU_COMPILATION_FLAGS}
+    BENCHMARK_TOOL
+      iree-benchmark-module
+    CONFIG
+      "${CONFIG}"
+    DRIVER
+      "${DRIVER}"
+    RUNTIME_FLAGS
+      "${RUNTIME_FLAGS}"
+  )
+endfunction(iree_benchmark_suite_x86)
 
-  BENCHMARK_MODES
-    "1-thread,full-inference,default-flags"
-  TARGET_BACKEND
-    "llvm-cpu"
-  TARGET_ARCHITECTURE
-    "CPU-x86_64-CascadeLake"
-  COMPILATION_FLAGS
-    ${LINUX_X86_64_CASCADELAKE_CPU_COMPILATION_FLAGS}
-  BENCHMARK_TOOL
-    iree-benchmark-module
-  CONFIG
-    "iree-llvm-cpu"
-  DRIVER
-    "local-task"
-  RUNTIME_FLAGS
-    "--task_topology_group_count=1"
-)
-
-# CPU, LLVM, local-task, 4 threads, x86_64, full-inference
-iree_benchmark_suite(
-  GROUP_NAME
-    "linux-x86_64"
-
-  MODULES
-    "${MINILM_L12_H384_UNCASED_INT32_SEQLEN128_MODULE}"
-    "${RESNET50_TF_FP32_MODULE}"
-    "${BERT_FOR_MASKED_LM_FP32_SEQLEN512_MODULE}"
-    "${EFFICIENTNET_V2_S_TF_FP32_MODULE}"
-
-  BENCHMARK_MODES
-    "4-thread,full-inference,default-flags"
-  TARGET_BACKEND
-    "llvm-cpu"
-  TARGET_ARCHITECTURE
-    "CPU-x86_64-CascadeLake"
-  COMPILATION_FLAGS
-    ${LINUX_X86_64_CASCADELAKE_CPU_COMPILATION_FLAGS}
-  BENCHMARK_TOOL
-    iree-benchmark-module
-  CONFIG
-    "iree-llvm-cpu"
-  DRIVER
-    "local-task"
-  RUNTIME_FLAGS
-    "--task_topology_group_count=4"
-)
-
-# CPU, LLVM, local-task, 8 threads, x86_64, full-inference
-iree_benchmark_suite(
-  GROUP_NAME
-    "linux-x86_64"
-
-  MODULES
-    "${MINILM_L12_H384_UNCASED_INT32_SEQLEN128_MODULE}"
-    "${RESNET50_TF_FP32_MODULE}"
-    "${BERT_FOR_MASKED_LM_FP32_SEQLEN512_MODULE}"
-    "${EFFICIENTNET_V2_S_TF_FP32_MODULE}"
-
-  BENCHMARK_MODES
-    "8-thread,full-inference,default-flags"
-  TARGET_BACKEND
-    "llvm-cpu"
-  TARGET_ARCHITECTURE
-    "CPU-x86_64-CascadeLake"
-  COMPILATION_FLAGS
-    ${LINUX_X86_64_CASCADELAKE_CPU_COMPILATION_FLAGS}
-  BENCHMARK_TOOL
-    iree-benchmark-module
-  CONFIG
-    "iree-llvm-cpu"
-  DRIVER
-    "local-task"
-  RUNTIME_FLAGS
-    "--task_topology_group_count=8"
-)
+# Produce all the configurations we care about.
+foreach(TARGET host CascadeLake)
```

Comment:
This needs to be configurable so that the benchmarks on CI don't become more expensive / higher latency due to having to also build and run the `host`  flavor.

To summarize:
 - CI needs to continue to build only CascadeLake
 - Local tests should probably default to `host` only.

Reply:
Good point!

Reply:
Done and I also refactored the patch to do more code sharing between tf and tflite (see the new linux-86_64-common.cmake file, which we may want to put somewhere else)